### PR TITLE
feature: add some utils that supernode will use

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

Before developing the supernode, some of the necessary tools and methods can help improve the development efficiency. And I wrote it based on the java version of supernode.

### Ⅱ. Does this pull request fix one issue?

None.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

Added.

### CODE REVIEW

File: common/util/net_util.go [hunk 1]
```diff
@@ -108,3 +112,80 @@ func ExtractHost(hostAndPort string) string {
...
 }
+
+// FilterParam filters request queries in URL.
+// Eg:
+// If you pass parameters as follows:
+//     url: http://a.b.com/locate?key1=value1&key2=value2&key3=value3
+//     filter: key2
+// and then you will get the following value as the return:
+//     http://a.b.com/locate?key1=value1&key3=value3
+func FilterParam(url, filter string) string {
```
Comment (lowzj) [inline common/util/net_util.go]: There is an implementation in package `dfget/core`: `getTaskURL`.

File: supernode/util/range_util.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	"strings"
+)
+
+const (
+	separator = "-"
+)
+
+// CalculatePieceNum calculates the number of piece
+// according to the parameter range.
+func CalculatePieceNum(rangeStr string) (pieceNum int) {
```
Comment (lowzj) [inline supernode/util/range_util.go]: The named return value `pieceNum` is not used in this function.

**Comment by Starnop** [comment]
All mentioned above has been updated. PTAL.

**Comment by lowzj** [comment]
LGTM

**Pull request merged**